modules/objdetect/src/qrcode.cpp

```diff
@@ -3810,33 +3831,502 @@ bool QRDetectMulti::computeTransformationPoints(const size_t cur_ind)
     return true;
 }
 
-bool QRCodeDetector::detectMulti(InputArray in, OutputArray points) const
-{
-    Mat inarr;
-    if (!checkQRInputImage(in, inarr))
-    {
-        points.release();
+struct QRCode;
+
+struct FinderPatternInfo {
+
+    FinderPatternInfo(const vector<Point2f>& patternPoints): points(patternPoints) {
+        float minSin = 1.f;
+        for (int i = 0; i < 4; i++) {
+            center += points[i];
+            const Point2f side = points[i]-points[(i+1) % 4];
+            const float lenSide = sqrt(normL2Sqr<float>(side));
+            minSin = min(minSin, abs(side.y) / lenSide);
+            moduleSize += lenSide;
+        }
+        moduleSize /= (4.f * 7.f); // 4 sides, 7 modules in one side
+        center /= 4.f;
+        minQrAngle = asin(minSin);
+        //std::cout << "minQrAngle " << minQrAngle << std::endl;
+        //std::cout << "minSin " << minSin << std::endl;
```

Comment:
Please use OpenCV logging if you plan to use it for further debugging.

Reply:
removed